ingest.py

```diff
@@ -1,28 +1,99 @@
+    texts = python_splitter.split_documents(documents_repo)
+    return texts
 
 def ingest_docs():
     """Get documents from web pages."""
-    loader = ReadTheDocsLoader("langchain.readthedocs.io/en/latest/")
-    raw_documents = loader.load()
-    text_splitter = RecursiveCharacterTextSplitter(
-        chunk_size=1000,
-        chunk_overlap=200,
-    )
-    documents = text_splitter.split_documents(raw_documents)
-    embeddings = OpenAIEmbeddings()
-    vectorstore = FAISS.from_documents(documents, embeddings)
 
-    # Save vectorstore
-    with open("vectorstore.pkl", "wb") as f:
-        pickle.dump(vectorstore, f)
+    urls = [
+        "https://api.python.langchain.com/en/latest/api_reference.html#module-langchain",
+        "https://python.langchain.com/docs/get_started", 
+        "https://python.langchain.com/docs/use_cases",
+        "https://python.langchain.com/docs/integrations",
+        "https://python.langchain.com/docs/modules", 
+        "https://python.langchain.com/docs/guides",
+        "https://python.langchain.com/docs/ecosystem",
+        "https://python.langchain.com/docs/additional_resources",
+        "https://python.langchain.com/docs/community",
+    ]
+
+    documents = []
+    for url in urls:
+        loader = RecursiveUrlLoader(url=url, max_depth=2 if url == urls[0] else 8, extractor=lambda x: Soup(x, "lxml").text, prevent_outside=True)
+        temp_docs = loader.load()
+        temp_docs = [doc for i, doc in enumerate(temp_docs) if doc not in temp_docs[:i]]        
+        documents += temp_docs
+        print("Loaded", len(temp_docs), "documents from", url)
+    
+    print("Loaded", len(documents), "documents from all URLs")
+    
+    html2text = Html2TextTransformer()
+    docs_transformed = html2text.transform_documents(documents)
+    
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=2000, chunk_overlap=200)
+    print('before splitting', len(docs_transformed))
+    docs_transformed = text_splitter.split_documents(docs_transformed)
+    print('after splitting', len(docs_transformed))
+    
+    repo_docs = ingest_repo()
+    docs_transformed += repo_docs
+        
+    # OPTION TO PICKLE
+    import pickle
+    with open('docs_transformed.pkl', 'wb') as f:
+        pickle.dump(docs_transformed, f)
+        
+    # with open('docs_transformed.pkl', 'rb') as f:
+    #     docs_transformed = pickle.load(f)
+    
+    client = weaviate.Client(url=WEAVIATE_URL, auth_client_secret=weaviate.AuthApiKey(api_key=WEAVIATE_API_KEY))
+    client.schema.delete_class("LangChain_newest_idx") # delete the class if it already exists
+
+    embeddings = OpenAIEmbeddings(chunk_size=200) # rate limit
 
+    batch_size = 100 # to handle batch size limit 
+    for i in range(0, len(docs_transformed), batch_size):
+        batch = docs_transformed[i:i+batch_size]
+        Weaviate.from_documents(batch, embeddings, client=client, by_text=False, index_name="LangChain_newest_idx")
```

Comment:
you should just be able to use `add_documents`

if you cant, thats a problem

Reply:
yep changed